pkgs/data/themes/adi1090x-plymouth-themes/shas.nix

```diff
@@ -0,0 +1,322 @@
+{
+  "abstract_ring" = {
```

Comment:
If we are downloading so many files, why not download the entire repo and unpack the archives in it?

Reply:
Hi @SuperSandro2000!  Thanks for taking the time to review my PR 😃 This is my first contribution to NixOS and I'm really excited to be participating.

I would expect the typical user will only install one of these at a time as plymouth only supports a single boot animation to be configured.  If we download the entire repo the user will wind up with 500+mb of unused plymouth themes in their nix store.  This seems like unwanted behavior.

This approach was adapted from the nerdfonts pkg below as it has a similar problem (collection is huge, most users only want one or two).

https://github.com/NixOS/nixpkgs/tree/nixos-22.11/pkgs/data/fonts/nerdfonts

If this isn't an acceptable pattern I could shift to using a different output for each (similar to the catppuccin-cursors pkg), but the concern here is there are 80+ themes, that would be a lot of outputs...

Reply:
Oh, right. I was assuming we could just download the correct theme on demand but that wouldn't cache well. 

I just noticed another thing: all the archives are downloaded from master, so the hashes will be invalid if one gets updates.

Reply:
Ahh great find.  Modified the script to retrieve the last master ref and make all calls using the ref-specific endpoints.  This should  guarantee that the shas.nix will always be accurate regardless if upstream has been updated and the package has not.